File: cpp_pdfwidget/pdfwidget.cpp
#include "pdfwidget.h"
#include <poppler-qt4.h>
#include <QAction>
#include <QPixmap>

PdfWidget::PdfWidget(QString path, QWidget *parent)
		: QLabel(parent), path(path)
{
	doc = Poppler::Document::load(path);
	doc->setRenderHint(Poppler::Document::TextAntialiasing);
	n_pages = doc->numPages();
	current_page = 0;
	load_current_page();

	QAction *next_page = new QAction("Next Page", this);
	next_page->setShortcut(Qt::Key_Right);
	connect(next_page, SIGNAL(triggered()), this, SLOT(next_page()));
	addAction(next_page);

	QAction *prev_page = new QAction("Prev Page", this);
	prev_page->setShortcut(Qt::Key_Left);
	connect(prev_page, SIGNAL(triggered()), this, SLOT(prev_page()));
	addAction(prev_page);
}

void PdfWidget::next_page()
{
	current_page++;
	if(current_page >= n_pages) current_page = 0;
	load_current_page();
}

void PdfWidget::prev_page()
{
	if(current_page) current_page--;
	else current_page = n_pages - 1;
	load_current_page();
}

void PdfWidget::load_current_page()
{
	setPixmap(QPixmap::fromImage(doc->page(current_page)->renderToImage()));
	setWindowTitle(QString("%1/%2 - %3").arg(current_page+1)
	                                    .arg(n_pages).arg(path));
}

File: pdf_widget.py
import sys

from PyQt4 import QtGui
from PyQt4 import QtCore
from popplerqt4 import Poppler

class Widget(QtGui.QLabel):
    def __init__(self, path):
        super().__init__()
        self.path = path
        self.doc = Poppler.Document.load(path)
        self.doc.setRenderHint(Poppler.Document.TextAntialiasing)
        self.n_pages = self.doc.numPages()
        self.current_page = 0
        self.load_current_page()

        next_page = QtGui.QAction('Next Page', self)
        next_page.setShortcut(QtCore.Qt.Key_Right)
        next_page.triggered.connect(self.next_page)
        self.addAction(next_page)

        prev_page = QtGui.QAction('Previous Page', self)
        prev_page.setShortcut(QtCore.Qt.Key_Left)
        prev_page.triggered.connect(self.prev_page)
        self.addAction(prev_page)

    def load_current_page(self):
        self.setPixmap(QtGui.QPixmap.fromImage(
                self.doc.page(self.current_page).renderToImage()))
        self.setWindowTitle('{0}/{1} - {2}'.format(
                self.current_page+1, self.n_pages, self.path))

    @QtCore.pyqtSlot()
    def next_page(self):
        self.current_page += 1
        if self.current_page >= self.n_pages:
            self.current_page = 0
        self.load_current_page()

    @QtCore.pyqtSlot()
    def prev_page(self):
        if self.current_page:
            self.current_page -= 1
        else:
            self.current_page = self.n_pages - 1
        self.load_current_page()


if __name__ == '__main__':
    app = QtGui.QApplication(sys.argv)
    file_path = QtGui.QFileDialog.getOpenFileName(
            None, 'Open PDF', '', 'PDF (*.pdf)')
    w = Widget(file_path)
    w.resize(600, 800)
    w.show()
    sys.exit(app.exec_())
